plugins/dbt/database_credentials.go

```diff
@@ -0,0 +1,54 @@
+package dbt
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func DatabaseCredentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.DatabaseCredentials,
+		DocsURL: sdk.URL("https://docs.getdbt.com/docs/core/connect-data-platform/redshift-setup#password-based-authentication"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Host,
+				MarkdownDescription: "Host to connect to.",
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.Port,
+				MarkdownDescription: "Port used to connect to.",
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.User,
+				MarkdownDescription: "User to authenticate as.",
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to.",
+				Secret:              true,
+			},
+			{
+				Name:                fieldname.Database,
+				MarkdownDescription: "Database name to connect to.",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer:           importer.TryEnvVarPair(defaultEnvVarMapping),
+	}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"DBT_HOST":     fieldname.Host,
+	"DBT_PORT":     fieldname.Port,
+	"DBT_USER":     fieldname.User,
+	"DBT_PASSWORD": fieldname.Password,
+	"DBT_DB":       fieldname.Database,
+}
```

Comment:
From what I gather from the documentation, these environment variables are only used to be referenced in a configuration file, is that correct?

With these provisioned, a config file such as the one documented here: https://docs.getdbt.com/reference/dbt-jinja-functions/profiles-yml-context

is still necessary for the plugin to work. Is there some documentation that I have missed?

Reply:
I see that this is indeed what you are documenting in the MR description as well.

Rather than creating an additional requirement for the user to write their own config file, I would prefer if we were to provision a configuration file directly, using `provision.TempFile`, as in plugins such as `mysql` or `akamai`.

Reply:
Thank you for the suggestion. Users would still need to configure `profiles.yml` file as lot of other attributes can be tweaked in it. I will add changes to incorporate `provision.TempFile` and keep env variables as it is.

Reply:
I see, thank you for the clarification.

@libutcher do you know of any other plugins that have a similar behaviour? Asking you since you tested/documented their vast majority, I wonder if there is already any occurrence of a plugin that requires additional setup on top of the  1Password CLI-related stuff.

Reply:
Following up here, it doesn't look like there are any plugins with that behaviour. @guru-pochineni, are you still planning to add changes to incorporate provision.TempFile?

Reply:
Haven't got the chance to make changes. Let me take a look into this next weekend.